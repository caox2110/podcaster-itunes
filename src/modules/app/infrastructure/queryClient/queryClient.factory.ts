import { QueryClient } from 'react-query';

const QueryClientFactory = {
  makeQueryClient: (): QueryClient =>
    new QueryClient({
      defaultOptions: {
        queries: {
          cacheTime: 1000 * 60 * 60 * 24, // 24 hours,
          staleTime: 1000 * 60 * 60 * 24, // 24 hours,
          refetchOnWindowFocus: false,
        },
      },
    }),
};
export { QueryClientFactory };
