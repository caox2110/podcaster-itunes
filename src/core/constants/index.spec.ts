import { constants as AllAppConstants } from '.';
import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

const logSpy = jest.spyOn(console, 'warn');

const {
  app: { IS_DEV, MODE },
} = AllAppConstants;

beforeEach(() => {
  logSpy.mockImplementation(() => {});
});

afterEach(() => {
  logSpy.mockRestore();
});

describe('All app constant vars', () => {
  it('Should be getted all the vars for the app', () => {
    const constants = {
      app: appConstant,
      theme: themeConstant,
      url: urlConstant,
      queryClient: queryClientConstant,
    };

    expect(AllAppConstants).toStrictEqual(constants);
    expect(AllAppConstants).toMatchSnapshot();
  });
  it('Should be printed "Constants are {AllAppConstants}"', () => {
    if (IS_DEV) {
      console.warn(`Constants are ${JSON.stringify(AllAppConstants)}`);
    }
    expect(logSpy).toHaveBeenCalledTimes(0);
    // TODO: Improve the test for printed warn
    // expect(logSpy).toHaveBeenCalledWith(
    //   expect.stringContaining(`Constants are ${JSON.stringify(AllAppConstants)}`),
    // );
  });
  it('Should be printed "ENV is dev"', () => {
    if (IS_DEV) {
      console.warn(`ENV is ${JSON.stringify(MODE)}`);
    }
    expect(logSpy).not.toHaveBeenCalled();
  });
});
