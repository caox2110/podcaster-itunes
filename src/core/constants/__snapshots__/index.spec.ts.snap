// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All app constant vars Should be getted all the vars for the app 1`] = `
Object {
  "app": Object {
    "APP_NAME": "Podcaster",
    "IS_DEV": false,
    "IS_PROD": false,
    "MODE": "test",
  },
  "theme": Object {
    "EMOTION_CACHE_KEY": "podcast-css",
    "FONT_PRIMARY": "'DM Sans', sans-serif",
    "MODE": Object {
      "DARK": "dark",
      "LIGHT": "light",
    },
  },
  "url": Object {
    "allowOriginApi": "https://api.allorigins.win/get?url=",
    "endpoints": Object {
      "lookupUrl": "lookup",
      "rssPodcastUrl": "us/rss/toppodcasts/limit=100/genre=1310/json",
    },
    "preApi": "https://itunes.apple.com/",
  },
}
`;
